diff --git a/dom/media/CubebUtils.cpp b/dom/media/CubebUtils.cpp
index 2f8a07828e..d9a1226a27 100644
--- a/dom/media/CubebUtils.cpp
+++ b/dom/media/CubebUtils.cpp
@@ -46,6 +46,7 @@
 #include "audioipc2_client_ffi_generated.h"
 #include "audioipc2_server_ffi_generated.h"
 #include "mozilla/StaticPrefs_media.h"
+#include "MaskConfig.hpp"
 
 #if defined(ENABLE_TESTS) || defined(FUZZING)
 #  define ENABLE_MOCK_CUBEB 1
@@ -501,6 +502,8 @@ uint32_t PreferredSampleRate(bool aShouldResistFingerprinting) {
   if (sCubebForcedSampleRate) {
     return sCubebForcedSampleRate;
   }
+  if (auto value = MaskConfig::GetUint32("AudioContext:sampleRate"))
+    return value.value();
   if (aShouldResistFingerprinting) {
     return 44100;
   }
diff --git a/dom/media/moz.build b/dom/media/moz.build
index 39b45f6b78..4fec92a0fb 100644
--- a/dom/media/moz.build
+++ b/dom/media/moz.build
@@ -417,5 +417,7 @@ if CONFIG["CC_TYPE"] in ("clang", "gcc"):
 CXXFLAGS += ["-Werror=switch"]
 
 FINAL_LIBRARY = "xul"
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
 
 SPHINX_TREES["/media"] = "docs"
diff --git a/dom/media/webaudio/AudioContext.cpp b/dom/media/webaudio/AudioContext.cpp
index 6b1c4fb7d4..ecdef2e040 100644
--- a/dom/media/webaudio/AudioContext.cpp
+++ b/dom/media/webaudio/AudioContext.cpp
@@ -6,6 +6,7 @@
 
 #include "AudioContext.h"
 
+#include "MaskConfig.hpp"
 #include "AudioBuffer.h"
 #include "AudioBufferSourceNode.h"
 #include "AudioChannelService.h"
@@ -556,6 +557,8 @@ double AudioContext::OutputLatency() {
   }
   // When reduceFingerprinting is enabled, return a latency figure that is
   // fixed, but plausible for the platform.
+  if (auto value = MaskConfig::GetDouble("AudioContext:outputLatency"))
+    return value.value();
   double latency_s = 0.0;
   if (mShouldResistFingerprinting) {
 #ifdef XP_MACOSX
@@ -709,6 +712,8 @@ void AudioContext::UnregisterActiveNode(AudioNode* aNode) {
 }
 
 uint32_t AudioContext::MaxChannelCount() const {
+  if (auto value = MaskConfig::GetUint32("AudioContext:maxChannelCount"))
+    return value.value();
   if (mShouldResistFingerprinting) {
     return 2;
   }
diff --git a/dom/media/webaudio/moz.build b/dom/media/webaudio/moz.build
index a2fe948600..497a63894a 100644
--- a/dom/media/webaudio/moz.build
+++ b/dom/media/webaudio/moz.build
@@ -154,3 +154,9 @@ include("/ipc/chromium/chromium-config.mozbuild")
 
 FINAL_LIBRARY = "xul"
 LOCAL_INCLUDES += [".."]
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
+
+# AudioFingerprintManager in dom/base
+LOCAL_INCLUDES += ["/dom/base"]
