diff --git a/netwerk/protocol/http/HttpBaseChannel.cpp b/netwerk/protocol/http/HttpBaseChannel.cpp
index 4132452313..961ace0b48 100644
--- a/netwerk/protocol/http/HttpBaseChannel.cpp
+++ b/netwerk/protocol/http/HttpBaseChannel.cpp
@@ -39,6 +39,7 @@
 #include "mozilla/browser/NimbusFeatures.h"
 #include "mozilla/dom/BrowsingContext.h"
 #include "mozilla/dom/CanonicalBrowsingContext.h"
+#include "mozilla/dom/NavigatorManager.h"
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/FetchPriority.h"
 #include "mozilla/dom/LoadURIOptionsBinding.h"
@@ -526,12 +527,29 @@ NS_IMETHODIMP
 HttpBaseChannel::SetDocshellUserAgentOverride() {
   RefPtr<dom::BrowsingContext> bc;
   MOZ_ALWAYS_SUCCEEDS(mLoadInfo->GetBrowsingContext(getter_AddRefs(bc)));
-  if (!bc) {
-    return NS_OK;
+  nsAutoString customUserAgent;
+  if (bc) {
+    bc->GetCustomUserAgent(customUserAgent);
+  }
+
+  if (customUserAgent.IsEmpty() || customUserAgent.IsVoid()) {
+    MOZ_ALWAYS_SUCCEEDS(
+        mLoadInfo->GetWorkerAssociatedBrowsingContext(getter_AddRefs(bc)));
+    if (bc) {
+      bc->GetCustomUserAgent(customUserAgent);
+      if (customUserAgent.IsEmpty() || customUserAgent.IsVoid()) {
+        dom::NavigatorManager::GetUserAgent(
+            bc->OriginAttributesRef().mUserContextId, customUserAgent);
+      }
+    }
+  }
+
+  if (customUserAgent.IsEmpty() || customUserAgent.IsVoid()) {
+    mozilla::OriginAttributes attrs;
+    MOZ_ALWAYS_SUCCEEDS(mLoadInfo->GetOriginAttributes(&attrs));
+    dom::NavigatorManager::GetUserAgent(attrs.mUserContextId, customUserAgent);
   }
 
-  nsAutoString customUserAgent;
-  bc->GetCustomUserAgent(customUserAgent);
   if (customUserAgent.IsEmpty() || customUserAgent.IsVoid()) {
     return NS_OK;
   }
diff --git a/netwerk/protocol/http/moz.build b/netwerk/protocol/http/moz.build
index d2330003ca..a5ea856173 100644
--- a/netwerk/protocol/http/moz.build
+++ b/netwerk/protocol/http/moz.build
@@ -246,3 +246,6 @@ XPCOM_MANIFESTS += [
 ]
 
 include("/tools/fuzzing/libfuzzer-config.mozbuild")
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 83a9dcff01..e5c881b8b0 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -14,6 +14,7 @@
 #include "nsError.h"
 #include "nsHttp.h"
 #include "nsHttpHandler.h"
+#include "MaskConfig.hpp"
 #include "nsHttpChannel.h"
 #include "nsHTTPCompressConv.h"
 #include "nsHttpAuthCache.h"
@@ -966,6 +967,15 @@ uint8_t nsHttpHandler::UrgencyFromCoSFlags(uint32_t cos,
 //-----------------------------------------------------------------------------
 
 const nsCString& nsHttpHandler::UserAgent(bool aShouldResistFingerprinting) {
+  if (auto value = MaskConfig::GetString("headers.User-Agent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+  if (auto value = MaskConfig::GetString("navigator.userAgent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+
   if (aShouldResistFingerprinting && !mSpoofedUserAgent.IsEmpty()) {
     LOG(("using spoofed userAgent : %s\n", mSpoofedUserAgent.get()));
     return mSpoofedUserAgent;
@@ -2082,6 +2092,10 @@ nsresult nsHttpHandler::SetAcceptLanguages() {
   }
 
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetString("headers.Accept-Language")) {
+    mAcceptLanguages.Assign(nsCString(value.value().c_str()));
+    return NS_OK;
+  }
 
   mAcceptLanguagesIsDirty = false;
 
@@ -2098,6 +2112,10 @@ nsresult nsHttpHandler::SetAcceptLanguages() {
 
 nsresult nsHttpHandler::SetAcceptEncodings(const char* aAcceptEncodings,
                                            bool isSecure, bool isDictionary) {
+  if (auto value = MaskConfig::GetString("headers.Accept-Encoding")) {
+    mHttpsAcceptEncodings.Assign(nsCString(value.value().c_str()));
+    return NS_OK;
+  }
   if (isDictionary) {
     mDictionaryAcceptEncodings = aAcceptEncodings;
   } else if (isSecure) {
