diff --git a/browser/app/firefox.exe.manifest b/browser/app/firefox.exe.manifest
index 995b2fc869..4a03da3064 100644
--- a/browser/app/firefox.exe.manifest
+++ b/browser/app/firefox.exe.manifest
@@ -3,10 +3,10 @@
 <assemblyIdentity
         version="1.0.0.0"
         processorArchitecture="*"
-        name="Firefox"
+        name="Camoufox"
         type="win32"
 />
-<description>Firefox</description>
+<description>Camoufox</description>
 <dependency>
         <dependentAssembly>
                 <assemblyIdentity
